src/v/storage/batch_cache.cc

```diff
@@ -361,6 +364,30 @@ void batch_cache_index::truncate(model::offset offset) {
     }
 }
 
+void batch_cache::background_reclaimer::start() {
+    _reclaim_loop = ss::with_scheduling_group(
+      _sg, [this] { return reclaim_loop(); });
+}
+ss::future<> batch_cache::background_reclaimer::stop() {
+    _stopped = true;
+    _change.signal();
+    return std::move(_reclaim_loop);
```

Comment:
@Lazin built a gate utility designed for co-routines. Storing the future here and returning it in stop is similar to a gate that you're trying to achieve so stop waits on background work?

Reply:
yeah, I will use gate for better readability.